seed/utils/match.py

```diff
@@ -0,0 +1,155 @@
+# !/usr/bin/env python
```

Comment:
Much of the methods were moved from the matching file within the `data_importer` module. I took the approach of separating matching logic specific to incoming data and general matching logic that includes logic for existing data.